src/plugins/intel_cpu/src/nodes/executors/acl/acl_fullyconnected.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "acl_fullyconnected.hpp"
+#include "acl_utils.hpp"
+#include "nodes/executors/executor.hpp"
+#include "nodes/executors/memory_arguments.hpp"
+#include "utils/debug_capabilities.h"
+#include "nodes/executors/debug_messages.hpp"
+#include "nodes/executors/implementation_utils.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+static MemoryPtr prepareWeightMemory(const MemoryPtr weightsMemory,
+                                     const ExecutorContext::CPtr context,
+                                     const ov::element::Type inPrec,
+                                     const ACLTensorAttrs& aclTensorAttrs) {
+    DEBUG_LOG("ACLFullyConnectedExecutor: prepack weights");
+    const auto& wgtDims = weightsMemory->getStaticDims();
+    const auto N = wgtDims[0];
+    const auto K = wgtDims[1];
+
+    auto create = [&]() {
+        MemoryPtr _ptr = std::make_shared<Memory>(context->getEngine(), CpuBlockedMemoryDesc(inPrec, weightsMemory->getShape()));
+        ACLCommonExecutor::weightsConvertPrecision(weightsMemory, _ptr, aclTensorAttrs);
+        DEBUG_LOG("ACLFullyConnectedExecutor: cache miss, perform packing");
+        return _ptr;
+    };
+
+    auto weightCache = context->getWeightsCache();
+    if (weightCache != nullptr) {
+        std::string format = "fc_acl_" + std::to_string(N) + "_" + std::to_string(K);
+        const std::string string_hash = format + "_" + std::to_string(weightsMemory->getSize()) + "_" +
+                                        std::to_string(reinterpret_cast<uint64_t>(weightsMemory->getData()));
+        DEBUG_LOG("ACLFullyConnectedExecutor: findOrCreate, string_hash: ", string_hash);
+        return *weightCache->findOrCreate(string_hash, create);
+    }
+
+    DEBUG_LOG("ACLFullyConnectedExecutor: Weights cache is not available");
+    return create();
+}
+
+ACLFullyConnectedExecutor::ACLFullyConnectedExecutor(const FCAttrs &attrs, const PostOps &postOps,
+                                                     const MemoryArgs &memory,
+                                                     const ExecutorContext::CPtr context) {
+    aclTensorAttrs.hasLayoutTypeNHWC = memory.at(ARG_SRC)->getDescPtr()->hasLayoutType(LayoutType::nspc);
+    fullyConnectedLayerInfo.weights_trained_layout = getAclDataLayoutByMemoryDesc(memory.at(ARG_WEI)->getDescPtr());
+    fullyConnectedLayerInfo.transpose_weights = !attrs.weightsNonTransposed;
```

Comment:
Does it mean ACL is going to execute transpose as part of fc execution?
If so, we should avoid this and prepare weights on compile time.